src/azure-cli/azure/cli/command_modules/batch/_params.py

```diff
@@ -71,15 +72,44 @@ def load_arguments(self, _):
+
+    with self.argument_context('batch account network-profile show') as c:
+        c.argument('resource_group_name', resource_group_name_type, help='Name of the resource group. If not specified will display currently set account.', required=False)
+        c.argument('account_name', batch_name_type, options_list=('--name', '-n'), help='Name of the batch account to show. If not specified will display currently set account.', required=False)
+
+    with self.argument_context('batch account network-profile set') as c:
+        c.argument('resource_group_name', resource_group_name_type, help='Name of the resource group. If not specified will display currently set account.', required=False)
+        c.argument('account_name', batch_name_type, options_list=('--name', '-n'), help='Name of the batch account to show. If not specified will display currently set account.', required=False)
+        c.argument('profile', help="Network profile to set. Allowed values include: 'BatchAccount', 'NodeManagement'")
+        c.argument('default_action', help="Default action for endpoint access. It is only applicable when publicNetworkAccess is enabled. Possible values include: 'Allow', 'Deny'", arg_type=get_enum_type(EndpointAccessDefaultAction))
+
+    with self.argument_context('batch account network-profile network-rule list') as c:
+        c.argument('resource_group_name', resource_group_name_type, help='Name of the resource group. If not specified will display currently set account.', required=False)
+        c.argument('account_name', batch_name_type, options_list=('--name', '-n'), help='Name of the batch account to show. If not specified will display currently set account.', required=False)
+
+    with self.argument_context('batch account network-profile network-rule add') as c:
+        c.argument('resource_group_name', resource_group_name_type, help='Name of the resource group. If not specified will display currently set account.', required=False)
+        c.argument('account_name', batch_name_type, options_list=('--name', '-n'), help='Name of the batch account to show. If not specified will display currently set account.', required=False)
+        c.argument('profile', help="Network profile to set. Allowed values include: 'BatchAccount', 'NodeManagement'")
+        c.argument('ip_address', help='IPv4 address or CIDR range.')
+
+    with self.argument_context('batch account network-profile network-rule delete') as c:
+        c.argument('resource_group_name', resource_group_name_type, help='Name of the resource group. If not specified will display currently set account.', required=False)
+        c.argument('account_name', batch_name_type, options_list=('--name', '-n'), help='Name of the batch account to show. If not specified will display currently set account.', required=False)
+        c.argument('profile', help="Network profile to set. Allowed values include: 'BatchAccount', 'NodeManagement'")
```

Comment:
Same as above. And I suggest that this argument type can be abstracted into `CLIArgumentType` to facilitate multiple reuse

Reply:
enum define and updated all the profile args to use a common CLIArgumentType